### PR TITLE
Update error formatting in pos process block

### PR DESCRIPTION
**Pull request by lazynina**

### CODE REVIEW

**Comment by lazynina** [comment]
Current dependencies on/for this PR:
* feature/pos-process-block
  * **PR #727** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/727" target="_blank">Graphite</a> 
    * **PR #728** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/728" target="_blank">Graphite</a> 
      * **PR #729** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/729" target="_blank">Graphite</a> 
        * **PR #730** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/730" target="_blank">Graphite</a> 
          * **PR #732** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/732" target="_blank">Graphite</a> 
            * **PR #733** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/733" target="_blank">Graphite</a> 
              * **PR #734** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/734" target="_blank">Graphite</a> 
                * **PR #735** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/735" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/735?utm_source=stack-comment).

File: lib/pos_blockchain.go [hunk 1]
```diff
@@ -83,28 +83,28 @@ func (bc *Blockchain) processBlockPoS(desoBlock *MsgDeSoBlock, currentView uint6
...
-		return false, false, nil, err
+		return false, false, nil, errors.Wrap(err, "processBlockPoS: Problem creating new block node")
 	}
 	// 5. Add block to best chain.
 	bc.addBlockToBestChain(newBlockNode)
 
 	// 6. Commit grandparent if possible.
 	if err = bc.commitGrandparents(desoBlock); err != nil {
-		return false, false, nil, err
+		return false, false, nil, errors.Wrap(err, "processBlockPoS: error committing grandparents")
```
Comment (tholonious) [inline lib/pos_blockchain.go:107]:
```suggestion
		return false, false, nil, errors.Wrap(err, "processBlockPoS: error committing grandparents: ")
```

File: lib/pos_blockchain.go [hunk 2]
```diff
@@ -83,28 +83,28 @@ func (bc *Blockchain) processBlockPoS(desoBlock *MsgDeSoBlock, currentView uint6
...
 	if err = bc.addBlockToBlockIndex(desoBlock); err != nil {
-		return false, false, nil, err
+		return false, false, nil, errors.Wrap(err, "processBlockPoS: Problem adding block to block index")
 	}
 
 	// 4. Handle reorgs if necessary
 	if bc.shouldReorg(desoBlock) {
 		if err = bc.handleReorg(desoBlock); err != nil {
-			return false, false, nil, err
+			return false, false, nil, errors.Wrap(err, "processBlockPoS: Problem handling reorg")
```
Comment (tholonious) [inline lib/pos_blockchain.go:92]:
```suggestion
			return false, false, nil, errors.Wrap(err, "processBlockPoS: Problem handling reorg: ")
```

File: lib/pos_blockchain.go [hunk 3]
```diff
@@ -44,20 +44,20 @@ func (bc *Blockchain) processBlockPoS(desoBlock *MsgDeSoBlock, currentView uint6
...
 
 	utxoView, err := bc.BuildUtxoViewToBlockParent(*desoBlock.Header.PrevBlockHash)
 	if err != nil {
-		return false, false, nil, errors.Wrapf(err, "processBlockPoS: Problem initializing UtxoView")
+		return false, false, nil, errors.Wrap(err, "processBlockPoS: Problem initializing UtxoView")
 	}
 	validatorsByStake, err := utxoView.GetFullSnapshotValidatorSetEntriesByStake()
 	if err != nil {
-		return false, false, nil, errors.Wrapf(err, "processBlockPoS: Problem getting validator set")
+		return false, false, nil, errors.Wrap(err, "processBlockPoS: Problem getting validator set")
```
Comment (tholonious) [inline lib/pos_blockchain.go:56]:
```suggestion
		return false, false, nil, errors.Wrap(err, "processBlockPoS: Problem getting validator set: ")
```

File: lib/pos_blockchain.go [hunk 4]
```diff
@@ -44,20 +44,20 @@ func (bc *Blockchain) processBlockPoS(desoBlock *MsgDeSoBlock, currentView uint6
...
 	// If so, we need to perform the rest of the validations and then add to our block index.
 	if err := bc.validateDeSoBlockPoS(desoBlock); err != nil {
-
+		return false, false, nil, errors.Wrap(err, "processBlockPoS: block validation failed: ")
 	}
 
 	utxoView, err := bc.BuildUtxoViewToBlockParent(*desoBlock.Header.PrevBlockHash)
 	if err != nil {
-		return false, false, nil, errors.Wrapf(err, "processBlockPoS: Problem initializing UtxoView")
+		return false, false, nil, errors.Wrap(err, "processBlockPoS: Problem initializing UtxoView")
```
Comment (tholonious) [inline lib/pos_blockchain.go:52]:
```suggestion
		return false, false, nil, errors.Wrap(err, "processBlockPoS: Problem initializing UtxoView: ")
```

**Review by tholonious — APPROVED** [review]
Looks good! AFAIK, whenever we use errors.Wrap, we add a colon at the end of the error message to separate our message from the error's body. Feel free to ignore my suggestions, if I'm wrong.

**Comment by lazynina** [comment]
### Merge activity

* **Oct 17, 7:42 PM**: @lazynina started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/735).

**Pull request merged**